File: algorithms/lab00/python/task-1/sol1_sort_asc.py
# SPDX-License-Identifier: BSD-3-Clause

class Solution:
    # Solutie: Sortam vectorul in ordine crescatoare si returnam al k-lea cel mai mare element.
    # Complexitate: O(nlogn)
    @staticmethod
    def findKthLargest(nums: list[int], k: int) -> int:
        nums.sort()
        return nums[-k]

File: algorithms/lab00/rust/task-1/sol1_sort_asc.rs
// SPDX-License-Identifier: BSD-3-Clause

// Solutie: Sortam vectorul in ordine crescatoare si returnam al k-lea cel mai mare element.
// Complexitate: O(n log n)

impl Solution {
    pub fn find_kth_largest(mut nums: Vec<i32>, k: i32) -> i32 {
        nums.sort();
        let idx = nums.len() - k as usize;
        nums[idx]
    }
}
